source/slang/slang-check-decl.cpp

```diff
@@ -7512,6 +7591,21 @@ namespace Slang
         }
         DeclAndCtorInfo structDeclInfo = DeclAndCtorInfo(m_astBuilder, this, structDecl, false);
 
+        // ensure all varDecl members are processed up to SemanticsBodyVisitor so we can be sure that if init expressions 
+        // of members are to be synthisised, they are.
+        bool isDefaultInitializableType = this->getShared()->findInheritance<InterfaceDecl>(structDecl, m_astBuilder->getDefaultInitializableType());
```

Comment:
The logic here means that we will not consider the type as default initializable iff there is no `:IDefaultInitializable` declared on the type. `extension MyType : IDefaultInitializable` won't work. If this is intended we probably need the logic to prevent extending a type that is not default initializable with some interface  that is default initializable.

Reply:
Are there any reasons why we can't allow `extension MyType : IDefaultIinitializable`?

Reply:
This is a bug with findInheritance. We should be checking for it. I will address this.

Reply:
In that case you can just call isSubtype or tryGetSubtypeWitness.

Reply:
I reread the plan for `IDefaultIinitializable`, I originally misunderstood what was the plan
`isSubtype` works well.